compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1125,19 +1119,31 @@ pub fn describe_flag_categories(early_dcx: &EarlyDiagCtxt, matches: &Matches) ->
     }
 
     if cg_flags.iter().any(|x| *x == "passes=list") {
-        let backend_name = debug_flags.iter().find_map(|x| x.strip_prefix("codegen-backend="));
-
-        let opts = config::Options::default();
-        let sysroot = filesearch::materialize_sysroot(opts.maybe_sysroot.clone());
-        let target = config::build_target_config(early_dcx, &opts, &sysroot);
-
-        get_codegen_backend(early_dcx, &sysroot, backend_name, &target).print_passes();
+        get_backend_from_raw_matches(early_dcx, matches).print_passes();
```

Comment:
Question: is it possible to add a test for this so we don't regress this in the future?

Reply:
@bjorn3 we run cg_clif’s tests on our CI, right? It may be easier to have a simple test there, than in-tree here (unless we have `//@ needs-cranelift` to execute your tests).

Reply:
I haven't changed cg_clif yet in the way I want precisely because the bug this PR fixes would prevent cargo from working.

Reply:
Hm, if this is *very* convoluted to write a test for, then I think it's acceptable to not have one (this is unbreak cg_clif in the first place)